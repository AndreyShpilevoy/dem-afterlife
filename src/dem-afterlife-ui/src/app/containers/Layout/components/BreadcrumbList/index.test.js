/* eslint no-undef: 0, fp/no-unused-expression: 0, fp/no-nil: 0, react/jsx-filename-extension:0 */

import React from 'react';
import {shallow} from 'enzyme';
import {BreadcrumbListPure} from './index';


jest.mock('../BreadcrumbListItem');

describe('BreadcrumbList Pure', () => {
    const classes = {
        list: 'list'
    };

    it('component should mach snapshot for 1 breadcrumb', () => {
        const breadcrumbArray = [{title: 'Conference', path: '/', order: 1}];
        expect(shallow(
            <BreadcrumbListPure
                classes={classes}
                breadcrumbArray={breadcrumbArray} />
        )).toMatchSnapshot();
    });

    it('component should mach snapshot for 4 breadcrumbs', () => {
        const breadcrumbArray = [
            {title: 'Conference 1', path: '/1', order: 1},
            {title: 'Conference 2', path: '/2', order: 2},
            {title: 'Conference 3', path: '/3', order: 3},
            {title: 'Conference 4', path: '/4', order: 4}
        ];
        expect(shallow(
            <BreadcrumbListPure
                classes={classes}
                breadcrumbArray={breadcrumbArray} />
        )).toMatchSnapshot();
    });

    it('component should mach snapshot for 6 breadcrumbs', () => {
        const breadcrumbArray = [
            {title: 'Conference 1', path: '/1', order: 1},
            {title: 'Conference 2', path: '/2', order: 2},
            {title: 'Conference 3', path: '/3', order: 3},
            {title: 'Conference 4', path: '/4', order: 4},
            {title: 'Conference 5', path: '/5', order: 5},
            {title: 'Conference 6', path: '/6', order: 6}
        ];
        expect(shallow(
            <BreadcrumbListPure
                classes={classes}
                breadcrumbArray={breadcrumbArray} />
        )).toMatchSnapshot();
    });
});
