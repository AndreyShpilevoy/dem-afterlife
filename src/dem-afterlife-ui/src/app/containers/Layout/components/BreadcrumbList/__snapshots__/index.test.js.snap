// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BreadcrumbList Pure component should mach snapshot for 1 breadcrumb 1`] = `
<div
  className="list"
>
  <Jss(BreadcrumbListItemPure)
    breadcrumb={
      Object {
        "order": 1,
        "path": "/",
        "title": "Conference",
      }
    }
    ellipsis={false}
    isActive={true}
  />
</div>
`;

exports[`BreadcrumbList Pure component should mach snapshot for 4 breadcrumbs 1`] = `
<div
  className="list"
>
  <Jss(BreadcrumbListItemPure)
    breadcrumb={
      Object {
        "order": 1,
        "path": "/1",
        "title": "Conference 1",
      }
    }
    ellipsis={false}
    isActive={false}
  />
  <Jss(BreadcrumbListItemPure)
    breadcrumb={
      Object {
        "order": 2,
        "path": "/2",
        "title": "Conference 2",
      }
    }
    ellipsis={false}
    isActive={false}
  />
  <Jss(BreadcrumbListItemPure)
    breadcrumb={
      Object {
        "order": 3,
        "path": "/3",
        "title": "Conference 3",
      }
    }
    ellipsis={false}
    isActive={false}
  />
  <Jss(BreadcrumbListItemPure)
    breadcrumb={
      Object {
        "order": 4,
        "path": "/4",
        "title": "Conference 4",
      }
    }
    ellipsis={false}
    isActive={true}
  />
</div>
`;

exports[`BreadcrumbList Pure component should mach snapshot for 6 breadcrumbs 1`] = `
<div
  className="list"
>
  <Jss(BreadcrumbListItemPure)
    breadcrumb={
      Object {
        "order": 1,
        "path": "/1",
        "title": "Conference 1",
      }
    }
    ellipsis={false}
    isActive={false}
  />
  <Jss(BreadcrumbListItemPure)
    breadcrumb={
      Object {
        "order": 2,
        "path": "/2",
        "title": "Conference 2",
      }
    }
    ellipsis={true}
    isActive={false}
  />
  <Jss(BreadcrumbListItemPure)
    breadcrumb={
      Object {
        "order": 4,
        "path": "/4",
        "title": "Conference 4",
      }
    }
    ellipsis={false}
    isActive={false}
  />
  <Jss(BreadcrumbListItemPure)
    breadcrumb={
      Object {
        "order": 5,
        "path": "/5",
        "title": "Conference 5",
      }
    }
    ellipsis={false}
    isActive={false}
  />
  <Jss(BreadcrumbListItemPure)
    breadcrumb={
      Object {
        "order": 6,
        "path": "/6",
        "title": "Conference 6",
      }
    }
    ellipsis={false}
    isActive={true}
  />
</div>
`;
